### PR TITLE
Chapter 9 Advanced Login (RE)

### PR DESCRIPTION
**Pull request by Iskandar19**

### CODE REVIEW

**Comment by Iskandar19** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -4,16 +4,43 @@ def log_in user
...
 
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by id: user_id
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by id: user_id
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+      @current_user = user
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]: wrong indent

**Comment by Iskandar19** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**